app/models/miq_report/search.rb

```diff
@@ -36,24 +50,18 @@ def get_order_info
     return [false, nil] unless db_class.sortable?
     # Convert sort cols from sub-tables from the form of assoc_name.column to the form of table_name.column
     order = sortby.to_miq_a.collect do |c|
-      info = col_to_col_info(c)
-      return [false, nil] if info[:virtual_reflection] || info[:virtual_column]
-
-      if c.include?(".")
-        assoc, col = c.split(".")
-        sql_col = [get_sqltable(assoc), col].join(".")
-      else
-        sql_col = [db_class.table_name, c].join(".")
-      end
-      sql_col = "LOWER(#{sql_col})" if [:string, :text].include?(info[:data_type])
+      model, col = association_klass(c)
+      # only use attributes that are accessible directly (not through virtual association)
+      return [false, nil] if model.nil?
+      sql_col = model.arel_attribute(col)
+      # only use regular columns or virtual columns with arel defined
+      return [false, nil] if sql_col.nil?
+      sql_col = sql_col.lower if [:string, :text].include?(model.type_for_attribute(col).type)
       sql_col
-    end.join(",")
+    end
 
-    unless self.order.nil?
-      case self.order.downcase
-      when "ascending"  then order += " asc"
-      when "descending" then order += " desc"
-      end
+    if (order_op = self.order_op)
+      order = order.map { |col| col.send(order_op) }
     end
 
     [true, order]
```

Comment:
I think we can lose the boolean return value here now that `order` is an array: `nil` == "unable to order"`; `[]` == "no order to apply"